test/jdk/java/awt/Mixing/MixingOnDialog.java

```diff
@@ -27,208 +27,75 @@
+    static Dialog d;
+    static Button heavy;
 
-    private static void init()
-    {
+    private static void init() {
         // Create components
-        final Dialog d = new Dialog((Frame)null, "Button-JButton mix test");
-        final Button heavy = new Button("  Heavyweight Button  ");
+        d = new Dialog((Frame)null, "Button-JButton mix test");
+        heavy = new Button("  Heavyweight Button  ");
         final JButton light = new JButton("  LW Button  ");
 
         // Actions for the buttons add appropriate number to the test sequence
-        heavy.addActionListener(new java.awt.event.ActionListener()
-                {
-                    public void actionPerformed(java.awt.event.ActionEvent e) {
-                        heavyClicked = true;
-                    }
-                }
-                );
-
-        light.addActionListener(new java.awt.event.ActionListener()
-                {
-                    public void actionPerformed(java.awt.event.ActionEvent e) {
-                        lightClicked = true;
-                    }
-                }
-                );
+        light.addActionListener(e -> lightClicked = true);
 
         // Overlap the buttons
-        heavy.setBounds(30, 30, 200, 200);
-        light.setBounds(10, 10, 50, 50);
+        heavy.setBounds(230, 230, 200, 200);
+        light.setBounds(210, 210, 50, 50);
 
         // Put the components into the frame
         d.setLayout(null);
         d.add(light);
         d.add(heavy);
-        d.setBounds(50, 50, 400, 400);
+        d.setBounds(250, 250, 400, 400);
         d.setVisible(true);
+    }
 
+    public static void main(String[] args) throws InterruptedException, InvocationTargetException, AWTException {
+        SwingUtilities.invokeAndWait(MixingOnDialog::init);
 
-        Robot robot = Util.createRobot();
-        robot.setAutoDelay(20);
-
-        Util.waitForIdle(robot);
+        Robot robot = new Robot();
+        robot.setAutoDelay(50);
+        robot.waitForIdle();
+        robot.delay(500);
 
         // Move the mouse pointer to the position where both
         //    buttons overlap
         Point heavyLoc = heavy.getLocationOnScreen();
         robot.mouseMove(heavyLoc.x + 5, heavyLoc.y + 5);
 
         // Now perform the click at this point
-        robot.mousePress(InputEvent.BUTTON1_MASK);
-        robot.mouseRelease(InputEvent.BUTTON1_MASK);
-        Util.waitForIdle(robot);
+        robot.mousePress(InputEvent.BUTTON1_DOWN_MASK);
+        robot.mouseRelease(InputEvent.BUTTON1_DOWN_MASK);
+        robot.waitForIdle();
 
-        // If the buttons are correctly mixed, the test sequence
-        // is equal to the check sequence.
-        if (lightClicked == true) {
-            MixingOnDialog.pass();
-        } else {
-            MixingOnDialog.fail("The lightweight component left behind the heavyweight one.");
-        }
-    }//End  init()
+        SwingUtilities.invokeLater(() -> d.dispose());
```

Comment:
invokeLater might not be executed before end of the test, it will be run after the main method ends.